src/LanguageServer/Protocol/Handler/Diagnostics/AbstractDocumentPullDiagnosticHandler.cs

```diff
@@ -34,16 +34,16 @@ protected override ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagno
         //
         // Only consider open documents here (and only closed ones in the WorkspacePullDiagnosticHandler).  Each
         // handler treats those as separate worlds that they are responsible for.
-        var textDocument = context.TextDocument;
-        if (textDocument is null)
+        var identifier = GetTextDocumentIdentifier(diagnosticsParams);
+        if (identifier is null || context.Document is null)
         {
             context.TraceDebug("Ignoring diagnostics request because no text document was provided");
             return new([]);
         }
 
-        if (!context.IsTracking(textDocument.GetURI()))
```

Comment:
this was the issue - for non-unc paths it does a case sensitive comparison, so the URI created from the file path on disk did not match the request URI (vscode normalizes it).